src/samplers/cpu/cpi/mod.rs

```diff
@@ -91,17 +179,87 @@ impl Sampler for Cpi {
 
         let elapsed = (now - self.prev).as_secs_f64();
 
-        match self.group.read() {
-            Ok(counts) => {
-                for (metric, pc) in &mut self.counters {
-                    metric.set(now, elapsed, counts[&pc]);
+        let mut total_cycles = 0;
```

Comment:
This isn't very idiomatic. I'd prefer these variables are declared where they're defined. I'd expect you get a bunch of clippy errors for this...